### PR TITLE
Integrate i18n into flashlight-build

### PR DESCRIPTION
**Pull request by oxtoacart**

### CODE REVIEW

File: src/github.com/getlantern/i18n/.gitignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Compiled Object files, Static and Dynamic libs (Shared Objects)
```
Comment (oxtoacart) [inline src/github.com/getlantern/i18n/.gitignore:1]: The i18n code has already been reviewed, this just pulls it into flashlight-build.

File: src/github.com/getlantern/jibber_jabber/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+language: go
```
Comment (oxtoacart) [inline src/github.com/getlantern/jibber_jabber/.travis.yml:1]: jibber_jabber is a transitive dependency of i18n, no review needed.

File: src/github.com/getlantern/tarfs/tarfs.go [hunk 3]
```diff
@@ -91,6 +91,26 @@ func New(tarData []byte, local string) (*FileSystem, error) {
 	return fs, nil
 }
 
+// SubDir returns a FileSystem corresponding to the given directory in the
+// original FileSystem.
+func (fs *FileSystem) SubDir(dir string) *FileSystem {
```
Comment (oxtoacart) [inline src/github.com/getlantern/tarfs/tarfs.go:96]: New tarfs function to slice off a subdirectory as its own FileSystem.

File: src/github.com/getlantern/flashlight/flashlight.go [hunk 4]
```diff
@@ -231,9 +243,14 @@ func configureSystemTray() {
 	}
 	systray.SetIcon(icon)
 	systray.SetTooltip("Lantern")
-	quit := systray.AddMenuItem("Quit", "Quit Lantern")
+	show := systray.AddMenuItem(i18n.T("TRAY_SHOW_LANTERN"), i18n.T("SHOW"))
+	quit := systray.AddMenuItem(i18n.T("TRAY_QUIT"), i18n.T("QUIT"))
 	go func() {
-		<-quit.ClickedCh
-		os.Exit(0)
+		select {
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/flashlight.go]: This needs to happen inside a loop, otherwise only one command will ever be executed!

**Comment by myleshorton** [comment]
We also used to just use the translation files from lantern-ui, which is handy because then translators just have to worry about that single file when translating the Lantern client. Looks like it should be pretty easy to just set the directory to the same directory the UI is using, however, and to use the same format?

File: src/github.com/getlantern/flashlight/flashlight.go (hunk 4 above)
Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/flashlight.go]: Good catch, fixed!

**Comment by oxtoacart** [comment]
> We also used to just use the translation files from lantern-ui, which is handy because then translators just have to worry about that single file when translating the Lantern client. Looks like it should be pretty easy to just set the directory to the same directory the UI is using, however, and to use the same format?

We still use the translation files from lantern-ui - https://github.com/getlantern/flashlight-build/blob/issue2240-merge/src/github.com/getlantern/flashlight/ui/ui.go#L32

**Comment by myleshorton** [comment]
> We still use the translation files from lantern-ui - https://github.com/getlantern/flashlight-build/blob/issue2240-merge/src/github.com/getlantern/flashlight/ui/ui.go#L32

Awesome!

File: src/github.com/getlantern/flashlight/flashlight.go (hunk 4 above)
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/flashlight.go]: I don't see that fix. Maybe on another branch?

**Comment by myleshorton** [comment]
Also, the windows version of this is a little wonky, as it flashes the command prompt window each time you show Lantern. I can't remember offhand how to avoid that but will look. I think we're using some version of `start` from the command prompt to trigger that right? There's some tweak to be made there.

File: src/github.com/getlantern/flashlight/flashlight.go (hunk 4 above)
Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/flashlight.go]: Oops, too many branches!  Fixed in here now.

**Comment by oxtoacart** [comment]
> Also, the windows version of this is a little wonky, as it flashes the command prompt window each time you show Lantern. I can't remember offhand how to avoid that but will look.

The production binary built with crosscompile.bash doesn't have a command-prompt window, so we should be good there.  See the -Hwindowsgui linker flag, which creates a windows rather than a command-line executable - https://github.com/getlantern/flashlight-build/blob/valencia/crosscompile.bash#L12

**Comment by myleshorton** [comment]
> The production binary built with crosscompile.bash doesn't have a command-prompt window, so we should be good there. See the -Hwindowsgui linker flag, which creates a windows rather than a command-line executable - https://github.com/getlantern/flashlight-build/blob/valencia/crosscompile.bash#L12

I think it's actually the call that happens from the system tray to open the browser -- that's what flashes the command prompt.

**Comment by oxtoacart** [comment]
Oh weird, even with that flag I'm seeing a command-line window shown.  I bet that's because of this - https://github.com/skratchdot/open-golang/blob/master/open/exec_windows.go

**Comment by myleshorton** [comment]
Yup -- that's what I was looking for thanks. I'll open a ticket and fiddle.

**Comment by oxtoacart** [comment]
@myleshorton Thanks!

**Comment by oxtoacart** [comment]
@myleshorton The trayhost library uses native code to do it - https://github.com/cratonica/trayhost/blob/master/platform/windows/tray.c

**Comment by myleshorton** [comment]
Oh right @oxtoacart -- so it needs to be fixed in both places. This PR in general looks fine, although now it looks like it's getting merge conflicts. You want me to merge manually, or do you want to fix the conflicts or merge manually yourself?

**Comment by oxtoacart** [comment]
@myleshorton Thanks for reviewing.  I'll handle the merge.

**Comment by oxtoacart** [comment]
Redoing with a different pull.

**Pull request closed without merging**